### ISSUE TITLE
New Release Timeline

### ISSUE THREAD

**Issue by jeremylt** [libxsmm/libxsmm#855]
#850 introduced some breaking changes. I was wondering if you folks are preparing for a release soon - a release would sure help libCEED communicate dependency requirements more easily and libXSMM is our recommended dependency for efficient CPU computation in our library.

Thanks!

**Comment by alheinecke** [comment]
@jeremylt yes, we are cleaning up for a v2 release which breaks some of the API. We are getting close, I assume 2-3 months. You are right #850 got rid of the transitional _v2 names and made them default. 

We still want to incl some features smoothing rough edges, mainly kernels that return "NULL" today on certain arch (like SSE, BF16 for ASIMD, etc.). 

Perhaps some higher level API might still get removed, but the core functionality is now stable for v2.

**Comment by jeremylt** [comment]
Awesome, thanks for the info!

**Comment by mbanck** [comment]
Any update on this?

Also, it would be nice if your intermediate tags (like [1.old_kernelapi_rip](https://github.com/libxsmm/libxsmm/releases/tag/1.old_kernelapi_rip) wouldn't lexically sort above your latest stable release (1.17)

**Comment by alheinecke** [comment]
We have still PPC64LE and system Z support pending. However, we are extensively CI/CD every commit on main and only PRs after review are merged. Is a version for you essential? Otherwise, please just pick head of main.

**Comment by jeremylt** [comment]
It's definitely still the case that 
>  a release would sure help libCEED communicate dependency requirements more easily and libXSMM is our recommended dependency for efficient CPU computation in our library.

**Comment by jeremylt** [comment]
Is libxsmm moving away from making any more releases? While that would not be my preference, it would be handy to know if that is libxsmm's plan so I can point our users at a specific commit hash as the dependency version instead of a version number

**Comment by alheinecke** [comment]
you can use any main hash as these are heavily tested (hence main moves slowly). We expect v2 when we have ACE support: https://x86ecosystem.org/wp-content/uploads/2026/03/ACE-Whitepaper-v1.pdf

**Comment by jeremylt** [comment]
Understood 

I get why years between releases is becoming normal for us scientific software developers with us all shretched so thin, but 'use LIBXSMM main and hope there isn't a bug in how we (libCEED) are using that dependency library' really isn't as comfortable of a thing for me to say to users as 'use version X of this dependency'.

**Comment by boegel** [comment]
I agree that it's a shame that there hasn't been a new release of libxsmm since v1.17 which dates from Dec'21 (!)...

So much development is not getting picked up broadly because of this, because many people strongly prefer sticking to "vetted" code, which is exactly what software releases are: the developers letting the community know that the software has improved significantly in some way, and is ready to be used widely.

"`main` is heavily tested" doesn't help most people, since they can't constantly keep an eye of what's changing in the main branch. Software releases are very helpful to let people know when it's interesting to update.

I've been hoping to pick up a new libxsmm release to employ it on Arm systems.
Right now, trying to build the latest release on Arm results in a clear error, though it seems to come from a different era:
```
error: #error Intel Architecture or compatible CPU required!
```

The time that most systems were running an Intel CPU is long behind is, you can't really ignore Arm anymore these days...

Waiting until specific features are implemented or stable is a never-ending story: most software is never done, but that doesn't mean you should never push out releases.

I'm fiddling with the `main` branch for now to explore things a bit on Arm, but I'm quite reluctant to use whatever random commit I happen to have picked up for any real usage...

Is there anything that the libxsmm users can do to help, like contributing to release notes, helping with automation to make pushing out releases easier, etc.?

**Comment by alheinecke** [comment]
For the “interesting update” point, we’re currently waiting for ACE support to be fully tested and merged.
The core developers are primarily employed by Intel, so x86 has naturally been the main focus so far. That said, we’d really love to see more involvement from the broader community—contributions are very welcome, and we encourage anyone interested to open RFCs or submit PRs.

At the moment, the release cycle is largely driven by x86 features, with upcoming major items including FP8 AMX and ACE support.
Regarding the ARM issue you’re seeing, it’s likely environment-specific. Our CI is currently green for ARM on Linux, ARM on macOS, and RV64 on Linux, please see `scripts/buildkite` to get an overview of all tests being run.

<img width="793" height="457" alt="Image" src="https://github.com/user-attachments/assets/bd7f83b2-e0e0-4c2f-8fa1-c9b9f67989cc" />

**Comment by boegel** [comment]
> For the “interesting update” point, we’re currently waiting for ACE support to be fully tested and merged. The core developers are primarily employed by Intel, so x86 has naturally been the main focus so far. That said, we’d really love to see more involvement from the broader community—contributions are very welcome, and we encourage anyone interested to open RFCs or submit PRs.
> 
> At the moment, the release cycle is largely driven by x86 features, with upcoming major items including FP8 AMX and ACE support. 

That's understandable, but that doesn't excuse the lack of a release since Dec'21 though...

> Regarding the ARM issue you’re seeing, it’s likely environment-specific. Our CI is currently green for ARM on Linux, ARM on macOS, and RV64 on Linux, please see `scripts/buildkite` to get an overview of all tests being run.

To be clear: I was (naively) trying the latest *release* (v1.17) on Arm, which probably doesn't have any Arm support at all yet.

With a very recent commit, I saw no problems on an NVIDIA Grace system (`make test` passed), except for some funky output at the end of the `scripts/tool_test.sh` script (the actual tests worked, but there were a bunch of errors on actions the script tried to take that utterly failed)

**Comment by Growl1234** [comment]
Thanks for the clarification. I share the downstream concerns already raised above, so I will not repeat them here.

My main concern is that making ACE a prerequisite leaves the release timeline rather open-ended, while v1.17 is already nearly five years old. Would you consider tagging the current state as "v2.0-rc1", "v1.99", or another clearly identified snapshot release, and then adding ACE in the final v2.0 or a subsequent v2.1 release?

Even an explicitly preliminary tag would be much easier for downstream packaging and dependency tracking than independently selecting arbitrary commits from "main".

**Comment by alheinecke** [comment]
I hope to have news on this in not so distant future. Right now I'm afraid of RPM packaging for redhat related packages as we heavily break API with latest release and refocus libxsmm to be a reference implementation of TPP https://arxiv.org/abs/2104.05755. This warrants 2.0 and we have still so major API change in the pipe due to low precision data types (MXFP8, MXFP4 etc.) which is planned to land with ACE changes. So please stay tuned for a little while, it will not be too long.

**Comment by alheinecke** [comment]
ACE was just merged today. In that process we also did extensive validation on ARM platforms. I'm currently working on cleaning up VS support under Windows and might fix some low hanging fruits wrt. raised issues. We plan to release 2.0 before/on June 30th. 
@Growl1234 : I see you just opened a PR. Do you plan more down the road and should we wait until mid July (we have a very short week in the US during the next week). That means for additional changes we would need ww28 stabilization meaning that the release would be on 7/13.

**Comment by Growl1234** [comment]
I am doing a few small CMake validates regarding this PR, but I didn't see how they materially affect the 2.0 release, so I think it can be proceeded as planned. Currently there are no follow-up plans regarding CMake build system.

Edit: The validation shows no problem to me.

**Comment by Growl1234** [comment]
I will deal with the [Spack recipe](https://github.com/spack/spack-packages/pull/5354) side.

One thing to confirm: would you plan to include `libxsmmConfig*.cmake` installed by GNU Makefile in the follow-up RPM builds?

**Comment by Growl1234** [comment]
It seems I was missing something. I will have a very small patch to make sure CMake reads version from current `version.txt`, rather than hardcode by itself.

**Comment by alheinecke** [comment]
https://github.com/libxsmm/libxsmm/releases/tag/2.0.0
